File: src/main/java/me/mrlion/lionplugin/listeners/BlockedCommandsListener.java
package me.mrlion.lionplugin.listeners;

import me.mrlion.lionplugin.LionPlugin;
import org.bukkit.ChatColor;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerCommandPreprocessEvent;

import java.util.List;
import java.util.Locale;
import java.util.Objects;

public class BlockedCommandsListener implements Listener {
    private final LionPlugin plugin;

    public BlockedCommandsListener(LionPlugin plugin) {
        this.plugin = plugin;
    }

    @EventHandler
    public void onCommandPreprocess(PlayerCommandPreprocessEvent event) {
        String message = event.getMessage().substring(1);
        String[] parts = message.split(" ");
        String commandName = parts[0].toLowerCase(Locale.getDefault());

        if (plugin.getConfig().contains("blocked-commands")) {
            List<String> blockedCommands = plugin.getConfig().getStringList("blocked-commands");

            if (blockedCommands.contains(commandName)) {
                if (!event.getPlayer().hasPermission("lionplugin.blockedcommands")) {
                    event.setCancelled(true);
                    String noPermsMessage = ChatColor.translateAlternateColorCodes('&', Objects.requireNonNull(plugin.getConfig().getString("blocked-commands-no-perms")));
                    event.getPlayer().sendMessage(noPermsMessage);
                }
            }
        }
    }
}

File: src/main/java/me/mrlion/lionplugin/listeners/BlockedCommandsListener.kt
package me.mrlion.lionplugin.listeners

import me.mrlion.lionplugin.LionPlugin
import org.bukkit.ChatColor
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.player.PlayerCommandPreprocessEvent
import java.util.*

class BlockedCommandsListener(private val plugin: LionPlugin) : Listener {
    @EventHandler
    fun onCommandPreprocess(event: PlayerCommandPreprocessEvent) {
        val message = event.message.substring(1)
        val parts = message.split(" ".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
        val commandName = parts[0].lowercase(Locale.getDefault())

        val blockedCommands = plugin.config.getStringList("blocked-commands")

        if (blockedCommands.contains(commandName)) {
            if (!event.player.hasPermission("lionplugin.blockedcommands")) {
                event.isCancelled = true
                val noPermsMessage = Objects.requireNonNull(plugin.config.getString("blocked-commands-no-perms"))?.let { ChatColor.translateAlternateColorCodes('&', it) }
                event.player.sendMessage(noPermsMessage)
            }
        }
    }
}
